frontend/tests/controllers-test/challengeCtrl.test.js

```diff
@@ -1,2445 +1,2445 @@
-                        data: 'error'
-                    });
-                }
-            };
-        });
-
-        it('change Baseline Status', function () {
-            var submissionId = 1;
-            expect(vm.baselineStatus).toEqual({});
-            vm.baselineStatus[1] = true;
-            vm.changeBaselineStatus(submissionId);
-            expect(vm.baselineStatus[submissionId]).toEqual(true);
-        });
-    });
-
-    describe('Unit tests for showRemainingSubmissions function \
-        `jobs/<challenge_id>/remaining_submissions`', function () {
-        var success, successResponse;
-        var errorResponse = {
-            error: 'error'
-        }
-
-        beforeEach(function () {
-            jasmine.clock().install;
-            spyOn(vm, 'startLoader');
-            spyOn(vm, 'stopLoader');
-            spyOn($rootScope, 'notify');
-
-            utilities.sendRequest = function (parameters) {
-                if (success) {
-                    parameters.callback.onSuccess({
-                        data: successResponse,
-                        status: 200
-                    });
-                } else {
-                    parameters.callback.onError({
-                        data: errorResponse
-                    });
-                }
-            };
-        });
-
-        afterEach(function () {
-            jasmine.clock().uninstall();
-        });
-
-        it('when submission limit exceeded', function () {
-            success = true;
-            successResponse = {
-                phases: {
-                    first_object: {
-                        id: undefined,
-                        limits: {
-                            submission_limit_exceeded: true,
-                            remaining_submissions_today_count: 0,
-                            remaining_time: 0,
-                            message: "Max Limit Exceeded",
-                        }
-                    },
-                }
-            };
-            vm.showRemainingSubmissions();
-            expect(vm.maxExceeded).toEqual(true);
-            expect(vm.maxExceededMessage).toEqual(successResponse.phases.first_object.limits.message);
-        });
-
-        it('when today remaining submission greater than 0', function () {
-            success = true;
-            successResponse = {
-                phases: {
-                    first_object: {
-                        id: undefined,
-                        limits: {
-                            submission_limit_exceeded: false,
-                            remaining_submissions_today_count: 10,
-                            remaining_time: 0,
-                        }
-                    }
-                }
-            };
-            vm.showRemainingSubmissions();
-            expect(vm.remainingSubmissions).toEqual(successResponse.phases.first_object.limits);
-            expect(vm.showSubmissionNumbers).toEqual(true);
-        });
-
-
```

Comment:
Can you please add only the necessary changes of this file?

Reply:
I am not sure why it displays so many lines changed, since I only edited one line: line 1780 (changed errorResponse to errorResponse.detail). I pulled Suryansh's branch from when he worked on this, and it still displays that many lines are changed, so I don't think that would be an issue when the code is merged with the master branch. 

The issue with visiting the challenge page was fixed. Here is a video showing the proper workflow: https://drive.google.com/file/d/1drKOBLkf7ObROC5fupGtc1_2r2JTRwTL/view?usp=sharing

Besides the tests, all of the changes are @Suryansh5545 's work, and his video showed the process working perfectly, so I assume the brief moment where the user is shown the participation page is due to lag on my machine.

Reply:
You can try merging with master first and then adding the additional lines. Please make changes only where required